### ISSUE TITLE
Parity 1.11 and UI 0.2.0 not working: the UI crashes.

### ISSUE THREAD

**Issue by gaia** [parity-js/shell#97]
_Before filing a new issue, please **provide the following information**._

> I'm running:
>
> - **Parity client version**: 1.11
> - **Parity UI version**: 0.2.0
> - **Operating system**: Windows (UI) / Linux (Node)
> - **Installed via**: installer / curl
> - **Fully synchronized**: yes
> - **Network**: ethereum
> - **I tried to restart both Parity UI and Parity Ethereum client**: yes

UI crashes.

Log:
```
2018-05-10 11:53:07  Starting Parity/v1.11.0-beta-62ccdd7ad-20180508/x86_64-windows-msvc/rustc1.25.0
2018-05-10 11:53:07  Keys path C:\Users\username\AppData\Roaming\Parity\Ethereum\keys\Foundation
2018-05-10 11:53:07  DB path C:\Users\username\AppData\Local\Parity\Ethereum\chains\ethereum\db\906a34e69aec8c0d
2018-05-10 11:53:07  Path to dapps C:\Users\username\AppData\Roaming\Parity\Ethereum\dapps
2018-05-10 11:53:07  State DB configuration: fast
2018-05-10 11:53:07  Operating mode: active
2018-05-10 11:53:07  Configured for Foundation using Ethash engine
2018-05-10 11:53:07  Unable to listen on 127.0.0.1:8546
WebSockets address 127.0.0.1:8546 is already in use, make sure that another instance of an Ethereum client is not running or change the address using the --ws-port and --ws-interface options.
2018-05-10 11:53:07  Public node URL: enode://195dcf8a4d83e7427e0f79a8711f909d8bd2a07c8f2b4bef9c620a8d99380908184989cf73bb7b9b882638cda8b932b311a80a38a5ce6ca521ae9fcece025385@127.0.0.1:30303
2018-05-10 11:53:07  Failed to auto-update latest ETH price: Fetch(Hyper(Io(Custom { kind: Other, error: Os { code: 10093, kind: Other, message: "Either the application has not called WSAStartup, or WSAStartup failed." } })))`

Tunnel line: `ssh -L 8180:localhost:8180 -L 8545:localhost:8545 -L 8546:localhost:8546 mynode
```

**Comment by Tbaut** [comment]
Could not reproduce.
@gaia it says that `WebSockets address 127.0.0.1:8546 is already in use`. Please make sure that you have only one instance of parity running.
This doesn't seem like a UI problem.

**Comment by gaia** [comment]
The parity node is remote. I uninstalled and re-installed the UI and the error persisted.
I don't even have parity installed on windows.

**Comment by Tbaut** [comment]
The node (on Linux) outputs this error, so please make sure no other application uses this port (8546)

**Comment by gaia** [comment]
Thibaut, my perception is that the UI tried to launch a local install of parity when it should not, as it was connected to the remote install. A blocks folder was created in

`C:\Users\user\AppData\Roaming\Parity`

When it was empty before. There were also two error dialogs that popped up. Would it help to post them here?

**Comment by Tbaut** [comment]
The UI tries to install parity if it can't reach any node on the localhost using the standard ports. It means that it failed to find the node you're tunnelling to. Please follow the recommendations here for the tunnel: https://wiki.parity.io/Wallet-Remote-Access.html `localhost` and `127.0.0.1` obviously aren't interpreted the same.

**Comment by amaurymartiny** [comment]
As @Tbaut said, Parity UI will check on startup if there's already a parity instance running (with a timeout). After that timeout, it'll run its own local instance of parity.

My guess is that because of the tunnel, the timeout passed, so UI started your local windows one.

I added a `--no-run-parity` flag in the latest v0.2.1, it will force UI **not** to run the local parity, even if no other parity instance is found. The timeout will hence be infinite, if it still doesn't work, then check the wiki link Thibaut posted.

**Comment by gaia** [comment]
I've been using the SSH tunnel since 1.6, never had a problem. If I run the UI without the flag `--no-run-parity` I get
https://i.imgur.com/ewWAA6p.png

Thank you for adding that flag, very useful! is it possible to use it via a conf file instead of the cli?

**Comment by gaia** [comment]
I was clicking on the default Parity UI shortcut.

I'm aware of --force-ui, but I want to start using the standalone UI because that's the path forward.

The JS error happened several times, both under 0.2.0 and 0.2.1. But I can't reproduce it now.

What is consistent:

- `--no-run-parity` works ok (some of the icons in the home screen are blanked out - they do not load)

- without the flag (straight parity.exe) I get the QUIT dialog, and the parity.exe.log shows the same info as in the original post, eventhough I've updated the tunnel to `ssh -L 8545:localhost:8545 -L 8546:localhost:8546 mynode` (removed 8180 tunneling)

Note: my parity node is not done syncing. I did DB kill, restarted with the config below, and it's been syncing for 10+ hrs

```
[parity]
base_path = "/mnt/lvm/ethereum/io.parity.ethereum"
db_path = "/mnt/lvm/ethereum/io.parity.ethereum/chains"
keys_path = "$HOME/.local/share/io.parity.ethereum/keys"
light = false

[network]
warp = true
min_peers = 20
max_peers = 50

[footprint]
tracing = "off"
fat_db = "off"
pruning = "fast"
pruning_history = 8
pruning_memory = 32
db_compaction = "ssd"

[snapshots]
disable_periodic = false

[ui]
force = false
```
